src/components/images/imageLoader.js

```diff
@@ -11,6 +12,85 @@ import 'css!./style';
         fillImageElement(elem, source);
     }
 
+    export function getImageBlurhashStr(hashes, tags) {
+        if (hashes && tags) {
+            return hashes[tags];
+        }
+        return null;
+    }
+
+    // function destroyBlurhash(target) {
+    //     let canvas = target.getElementsByClassName('blurhash-canvas')[0];
+    //     target.removeChild(canvas);
+    //     target.classList.remove('blurhashed');
+    // }
+
+    // function createBhashCanvas(blurhashstr, child, width, height, target) {
+        
+    // }
+
+    function itemBlurhashing(entry) {
+        // This intersection ratio ensures that items that are near the borders are also blurhashed, alongside items that are outside the viewport
+        // if (entry.intersectionRation <= 0.025)
+        if (entry.target) {
+            let target = entry.target;
+            // We only keep max 80 items blurhashed in screen to save memory
+            // if (document.getElementsByClassName('blurhashed').length <= 80) {
+
+            //} else {
+                // destroyBlurhash(target);
+            //}
+            let blurhashstr = target.getAttribute('data-blurhash');
+            if (blurhashstr && blurhash.isBlurhashValid(blurhashstr) && target.getElementsByClassName('blurhash-canvas').length === 0) {
+                console.log('Blurhashing item ' + target.parentElement.parentElement.parentElement.getAttribute('data-index') + ' with intersection ratio ' + entry.intersectionRatio);
+                let width = target.offsetWidth;
+                let height = target.offsetHeight;
```

Comment:
Part of the performance issues come from this.

You are decoding it to a full size, which is much more expensive.

Per the README of Blurhash:
```
Similarly, when displaying the placeholders, very small images work very well when scaled up. We usually decode placeholders that are 32 or even 20 pixels wide, and then let the UI layer scale them up, which is indistinguishable from decoding them at full size.
```

So the ideal implementation here would be to compute the aspect ratio of the element, then set the width to 32 and use the ratio to get the height of the element.

This should significantly reduce the cost of blurhash.

Reply:
Missed those bits. Let me test that out. Thank you very much!